developer-hub/src/reference_docs/glossary.rs

```diff
@@ -36,41 +36,81 @@
+//! without bloating the blockchain.
+//!
 //! #### Host Function:
 //!
+//! Host functions are functions provided by the runtime environment (the [host](#host)) to the Wasm
+//! runtime. These functions allow the Wasm code to interact with and perform operations on the
+//! [node](#node), like accessing the blockchain state.
+//!
 //! #### Runtime API:
 //!
+//! The runtime API acts as a communication bridge between the runtime and the node, serving as the
+//! exposed interface that facilitates their interactions.
+//!
 //! #### Dispatchable:
 //!
-//! Callable
+//! Dispatchables are functions that can be called by external entities, such as users or external
+//! systems, to interact with the blockchain's state. They are a core aspect of the runtime logic,
+//! handling transactions and other state-changing operations.
+//!
+//! **Synonyms**: Callable
 //!
 //! #### Extrinsic
 //!
+//! An extrinsic is a general term for a piece of data that is originated outside of the runtime and
+//! fed into the it as a part of the block-body. This includes user-initiated transactions as well
+//! as inherents which are placed into the block by the block-builder.
 //!
 //! #### Pallet
 //!
-//! #### Full Node
-//!
-//! #### Archive Node
-//!
-//! #### Validator
-//!
-//! #### Collator
-//!
-//! #### Parachain
-//!
-//! aka. AppChain.
-//!
+//! [FRAME](frame) pallets are modular components that encapsulate specific functionalities or
+//! business logic of a blockchain. They are the building blocks used to construct a blockchain's
+//! runtime, allowing for customizable and upgradeable networks. Pallets can be used to extend the
+//! capabilities of a Substrate-based blockchain in a composable way.
 //!
+//! #### Full Node
 //!
+//! It is a node that prunes historical states, keeping only recent finalized block states to reduce
+//! storage needs. It can potentially rebuild all states to become an archive node. Full nodes
+//! provide current chain state access and allow direct submission and validation of extrinsics,
+//! maintaining network decentralization.
 //!
+//! #### Archive Node
 //!
+//! An archive node is a specialized node that maintains a complete history of all block states and
+//! transactions. Unlike a full node, it does not prune historical data, ensuring full access to the
+//! entire blockchain history. This makes it essential for detailed blockchain analysis and
+//! historical queries, but requires significantly more storage capacity.
 //!
+//! #### Validator
 //!
+//! A validator is a node that participates in the consensus mechanism, validating transactions and
+//! blocks, and maintaining the integrity and security of the network.
 //!
+//! #### Collator
 //!
+//! A collator is a node that is responsible for producing candidate blocks for the validators.
+//! Collators are similar to validators on any other blockchain but, they do not need to provide
+//! security guarantees as the Relay Chain handles this.
 //!
+//! #### Parachain
```

Comment:
Might be cool to link to the parachain implementers guide here somewhere: https://paritytech.github.io/polkadot-sdk/book/index.html